src/StagedProposalProcessor.sol

```diff
@@ -525,16 +532,24 @@ contract StagedProposalProcessor is
             // If plugin proposal creation should be manual, skip it.
             if (plugin.isManual) continue;
 
-            bytes memory actionData = abi.encodeCall(
-                this.reportProposalResult,
-                (_proposalId, _stageId, plugin.resultType, stage.vetoThreshold == 0)
-            );
-
             Action[] memory actions = new Action[](1);
-            actions[0] = Action({to: address(this), value: 0, data: actionData});
+
+            bytes memory customParams = _stageId == 0 && _stageProposalParams.length > 0
+                ? _stageProposalParams[i]
+                : createProposalParams[_proposalId][_stageId][i];
```

Comment:
Even though it's an optimization, it's a bit strange to have two sources for the same parameters.
Isn't it possible to streamline things a bit more here?